consumer/pdata/common.go

```diff
@@ -391,13 +391,15 @@ func (am AttributeMap) InitFromMap(attrMap map[string]AttributeValue) AttributeM
 	return am
 }
 
-// InitEmptyWithCapacity constructs an empty AttributeMap with predefined slice capacity.
-func (am AttributeMap) InitEmptyWithCapacity(cap int) {
-	if cap == 0 {
-		*am.orig = []otlpcommon.KeyValue(nil)
+// EnsureCapacity increases the capacity of this AttributeMap instance, if necessary,
+// to ensure that it can hold at least the number of elements specified by the capacity argument.
+func (am AttributeMap) EnsureCapacity(capacity int) {
+	if capacity <= cap(*am.orig) {
 		return
 	}
-	*am.orig = make([]otlpcommon.KeyValue, 0, cap)
+	oldOrig := *am.orig
+	*am.orig = make([]otlpcommon.KeyValue, 0, capacity)
+	copy(*am.orig, oldOrig)
```

Comment:
Append may be faster to increase the capacity, assuming memory manager has a way to extend the allocated block without reallocation (but I did not verify): https://github.com/golang/go/wiki/SliceTricks#extend
It may be worth benchmarking.

Reply:
We can play with the internal implementation.